packages/terra-infinite-list/src/InfiniteList.jsx

```diff
@@ -0,0 +1,435 @@
+      this.updateHeight(node, index);
+      if (child.props.refCallback) {
+        child.props.refCallback(node);
+      }
+    };
+
+    let newProps = {};
+    if (this.props.isSelectable) {
+      const wrappedOnClick = SelectableUtils.wrappedOnClickForItem(child, index, this.props.onChange);
+      const wrappedOnKeyDown = SelectableUtils.wrappedOnKeyDownForItem(child, index, this.props.onChange);
+      newProps = SelectableUtils.newPropsForItem(child, index, wrappedOnClick, wrappedOnKeyDown, this.props.hasChevrons, this.props.selectedIndexes, this.props.disableUnselectedItems);
+    }
+
+    newProps.refCallback = wrappedCallBack;
+    newProps['data-infinite-list-index'] = index;
+    newProps.style = child.props.style ? Object.assign({}, child.props.style, { overflow: 'hidden' }) : { overflow: 'hidden' };
+    return React.cloneElement(child, newProps);
+  }
+
+  render() {
+    const {
+      children,
+      disableUnselectedItems,
+      hasChevrons,
+      initialLoadingIndicator,
+      isDivided,
+      isFinishedLoading,
+      isSelectable,
+      onRequestItems,
+      progressiveLoadingIndicator,
+      selectedIndexes,
+      ...customProps
+    } = this.props;
+    const topSpacer = createSpacer(`${this.boundary.hiddenTopHeight > 0 ? this.boundary.hiddenTopHeight : 0}px`, 0);
+    const bottomSpacer = createSpacer(`${this.boundary.hiddenBottomHeight > 0 ? this.boundary.hiddenBottomHeight : 0}px`, 1);
+
+    let loadingSpinner;
+    let visibleChildren;
+    let showDivided = isDivided;
+    if (!isFinishedLoading) {
+      if (this.childCount > 0) {
+        loadingSpinner = <List.Item content={progressiveLoadingIndicator} isSelectable={false} key={`infinite-spinner-row-${this.loadingIndex}`} />;
+      } else {
+        visibleChildren = <List.Item content={initialLoadingIndicator} isSelectable={false} key="infinite-spinner-full" style={{ height: '100%', position: 'relative' }} />;
+        showDivided = false;
+      }
+    }
+
+    let newChildren;
+    if (!visibleChildren) {
+      let upperChildIndex = this.lastChildIndex;
+      if ((!this.scrollGroups.length && this.lastChildIndex <= 0) || !this.renderNewChildren) {
+        upperChildIndex = this.childCount;
+      } else {
+        newChildren = (
+          <List {...customProps} isDivided={isDivided} className={cx(['infinite-hidden'])}>
+            {InfiniteUtils.getNewChildren(this.lastChildIndex, this.childrenArray, this.wrapChild)}
+          </List>
+        );
+        this.isRenderingNew = true;
+      }
+      visibleChildren = InfiniteUtils.getVisibleChildren(this.scrollGroups, this.childrenArray, this.boundary.topBoundryIndex, this.boundary.bottomBoundryIndex, this.wrapChild, upperChildIndex);
+    }
+
+    return (
+      <React.Fragment>
+        <List {...customProps} isDivided={showDivided} className={cx(['infinite-list', customProps.className])} refCallback={this.setContentNode}>
+          {topSpacer}
+          {visibleChildren}
```

Comment:
List inside of a list?

Reply:
Never mind. I see this is a list of visible and a list of hidden rendered together